drltrace/scripts/headers_parser.py

```diff
@@ -0,0 +1,176 @@
+    return final_str
+
+def parse_headers():
+    '''
+    @in - the routine parse SDKs headers located in HEADERS_PATH
+    @out - a list of lines from each header
+    '''
+
+    headers_content = list()
+    headers = [f for f in listdir(HEADERS_PATH) if isfile(join(HEADERS_PATH, f))]
+    for header in headers:
+        if not header.endswith(".h"):
+            continue
+        header_content = open(HEADERS_PATH + header, 'r').readlines()
+        headers_content.append(header_content)
+    return headers_content
+
+def find_in_headers(entry_name, headers_content):
+    '''
+    The function looks for the entry_name in the headers_count and returns a list of
+    potentially similar functions (that looks like WinAPI function prototype)
+    @in entry_name - a name of WinAPI function to search for
+    @in headers_content - headers content to search in
+    @out - a list of WinAPI function prototypes
+    '''
+
+    list_of_results = list()
+    for header_content in headers_content:
+        for idx, line in enumerate(header_content):
+            if entry_name in line:
+                ''' Parse line, look for separate names '''
+                if "(" in line and not "#define" in line and not "return " in line\
+                   and not "//" in line and not "/*" in line and not "*/" in line\
+                   and not "=" in line and not "STDMETHOD" in line\
+                   and not "EXTERN_GUID" in line:
+
+                    final_line = tuple()
+                    tmp_idx = idx
+                    #take return type, it can sit before function or at the same line
+                    while tmp_idx > 1:
+                        tmp_idx = tmp_idx - 1 # looking back
+                        if header_content[tmp_idx] == '\n' or ';' in header_content[tmp_idx]\
+                           or "#" in header_content[tmp_idx]:
+                            break
+                        if header_content[tmp_idx].startswith("\\"): # ignore comments
+                            continue
+                        final_line = final_line + (header_content[tmp_idx],)
+
+                    tmp_idx = idx
+                    # take arguments if exist
+                    while tmp_idx < len(header_content):
+                        if not header_content[tmp_idx].startswith("//"):
+                            comment_idx = header_content[tmp_idx].find("//")
+                            new_line = header_content[tmp_idx]
+                            if comment_idx: # remove comment after an argument
+                                new_line = header_content[tmp_idx][:comment_idx]
+                            final_line = final_line + (new_line,)
+                        if ";" in header_content[tmp_idx]:
+                            break
+                        tmp_idx = tmp_idx + 1
+                    # we have a limitation for the prototype length (e.g. no more than
+                    # 70 elements).
```

Comment:
Yes, sounds confusing. I found the threshold empirically for kernel32.dll but it might be different for other DLLs. The comment is changed.